src/main/java/com/faforever/client/teammatchmaking/PartyMemberItemController.java

```diff
@@ -0,0 +1,176 @@
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+import java.lang.ref.WeakReference;
+
+import static javafx.beans.binding.Bindings.createObjectBinding;
+import static javafx.beans.binding.Bindings.createStringBinding;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class PartyMemberItemController implements Controller<Node> {
+
+  private static final PseudoClass LEADER_PSEUDO_CLASS = PseudoClass.getPseudoClass("leader");
+  private static final PseudoClass PLAYING_PSEUDO_CLASS = PseudoClass.getPseudoClass("playing");
+
+  private final CountryFlagService countryFlagService;
+  private final AvatarService avatarService;
+  private final PlayerService playerService;
+  private final TeamMatchmakingService teamMatchmakingService;
+  private final UiService uiService;
+  private final ChatService chatService;
+  private final I18n i18n;
+
+  public Node playerItemRoot;
+  public ImageView avatarImageView;
+  public ImageView countryImageView;
+  public ImageView leagueImageView;
+  public Button kickPlayerButton;
+  public Label clanLabel;
+  public Label usernameLabel;
+  public Label leagueLabel;
+  public Label gameCountLabel;
+  public Label uefLabel;
+  public Label cybranLabel;
+  public Label aeonLabel;
+  public Label seraphimLabel;
+  public Label crownLabel;
+  public HBox playerCard;
+  public ImageView playerStatusImageView;
+
+  private Player player;
+  //TODO: this is a bit hacky
+  private WeakReference<ChatUserContextMenuController> contextMenuController = null;
+
+  @Override
+  public void initialize() {
+    clanLabel.managedProperty().bind(clanLabel.visibleProperty());
+  }
+
+  @Override
+  public Node getRoot() {
+    return playerItemRoot;
+  }
+
+  private ChatChannelUser chatUser;
+
+  public PartyMemberItemController(CountryFlagService countryFlagService, AvatarService avatarService, PlayerService playerService, TeamMatchmakingService teamMatchmakingService, UiService uiService, ChatService chatService, I18n i18n) {
+    this.countryFlagService = countryFlagService;
+    this.avatarService = avatarService;
+    this.playerService = playerService;
+    this.teamMatchmakingService = teamMatchmakingService;
+    this.uiService = uiService;
+    this.chatService = chatService;
+    this.i18n = i18n;
+  }
+
+  void setMember(PartyMember member) {
+    this.player = member.getPlayer();
+    //TODO: this is a bit hacky, a chat channel user is required to create a context menu as in the chat tab (for foeing/befriending/messaging people...)
+    chatUser = new ChatChannelUser(player.getUsername(), chatService.getChatUserColor(player.getUsername()), false, player);
+
+    countryImageView.imageProperty().bind(createObjectBinding(() -> countryFlagService.loadCountryFlag(
+        StringUtils.isEmpty(player.getCountry()) ? "" : player.getCountry()).orElse(null), player.countryProperty()));
```

Comment:
`StringUtils.isEmpty(player.getCountry()) ? "" : player.getCountry()`

If the string is empty it's empty? Thanks captain.

Reply:
This seems to be a nullcheck, no?

Reply:
Yes, but the CountryFlagService.loadCountryFlag has null handling builtin ^^